src/sinks/elasticsearch/config.rs

```diff
@@ -163,7 +163,7 @@ pub struct ElasticsearchConfig {
     #[serde(rename = "distribution")]
     pub endpoint_health: Option<HealthConfig>,
 
-    #[serde(alias = "normal", default)]
+    #[serde(alias = "normal", default = "default_bulk")]
     #[configurable(derived)]
     pub bulk: Option<BulkConfig>,
```

Comment:
I think if we go the route of defining a default during deserialization like this then `bulk` doesn't need to be an `Option`, it can just be a `BulkConfig` since it will always be `Some` (unless the user purposefully configures `null`, but that doesn't seem sensible to allow).

Reply:
Actually, I don't think we do want to do that since I remembered this option only applies if the `mode` is `bulk`. I'd suggest removing this default and adding back the `unwrap_or_else` that used to be there.

Reply:
However- the default mode is actually `bulk`. So I think it might actually make sense to have this default to `Some(BulkConfig::default())` .

If the user instead specified the mode to be datastream, this configuration would have no effect it seems.

Would that cause issues?

Reply:
Hmm, yeah, I guess that seems fine, but if the default is `Some(..)` then I feel like having it be an `Option` doesn't really make sense since I can't think of when a user would need to set it to `None` by setting `bulk: null` explicitly.

Reply:
Yeah good point. There are conflicting aspects of this. I prefer not having implied defaults. But now it seems weird to have it not be Option when it truly depends on `Mode`.

Maybe an alternative would be to have `ElasticsearchMode` enum variants contain the respective structs instead of having those as `Option<T>` at the root of the config.

Reply:
So that we're not refactoring this sink's entire config structure as part of a regression fix, I'll take the more minimal set of changes path and defer the improvements to the mode field for a later time.

Reply:
Agreed 👍